lib/src/signals/logic.dart

```diff
@@ -319,13 +319,59 @@ class Logic {
   void operator <=(Logic other) => gets(other);
 
   /// Logical bitwise NOT.
-  Logic operator ~() => NotGate(this).out;
+  Logic operator ~() {
+    if (this is Const) {
+      if (value.isValid) {
+        if (value == LogicValue.of(0, width: width)) {
+          return Const(LogicValue.filled(width, LogicValue.one));
+        } else if (value == LogicValue.filled(width, LogicValue.one)) {
+          return Const(LogicValue.of(0, width: width));
+        }
+      } else {
+        return Const(LogicValue.x, width: width);
+      }
+    }
+    return NotGate(this).out;
+  }
 
   /// Logical bitwise AND.
-  Logic operator &(Logic other) => And2Gate(this, other).out;
+  Logic operator &(Logic other) {
+    if (this is Const || other is Const) {
+      if (value.isValid && other.value.isValid) {
+        if (value == LogicValue.of(0, width: width) ||
+            other.value == LogicValue.of(0, width: width)) {
+          return Const(LogicValue.of(0, width: width));
+        } else if (value == LogicValue.filled(width, LogicValue.one)) {
+          return other;
```

Comment:
What if `other is Const` and `this` is not `Const` but has a value of all 1s currently?  Need to be cautious to only consider the value if it's a `Const`.